pkg/kata/pid_pool.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kata
+
+import (
+	"errors"
+	"os"
+	"sync"
+	"syscall"
+)
+
+type pidPool struct {
+	sync.Mutex
+	pool map[uint32]struct{}
+	cur  uint32
+}
+
+func newPidPool() *pidPool {
+	return &pidPool{
+		pool: make(map[uint32]struct{}),
+		cur:  uint32(os.Getpid()),
+	}
+}
+
+func (p *pidPool) Get() (uint32, error) {
```

Comment:
I thought VM would need a new pid. But recently I think KataContainers has solved it, and the previous design is not clear.

Reply:
Still I don't follow you. In kata containers, there is a kata-shim that can proxy signals and stdio for a container. The shim's pid is used as the container's pid on the host so that `kill` can send signals to the container process. That's why I asked how you plan to use this fake pid. Does containerd require such a valid host pid to represent the container process and use it to pass signals to the process, instead of calling `runtime kill` to pass the signals?

Reply:
Yep. What I mean is also that I can call the `KillContainer` API to pass the signals.

Reply:
OK, that look ok. Then please remove the unneeded pidpool.

Reply:
Unnecessary functions have been removed.